### ISSUE TITLE
Images disappear after editing new page

### ISSUE THREAD

**Issue by mikezimm** [mikezimm/LibraryCopier#9]
## This is the html structure of an embedded image in text.

Look at maybe finding the image tags and replacing them with new web parts instead.

![image](https://user-images.githubusercontent.com/49648086/170778683-8e08acd4-bc00-48df-82cf-d27bbffc3ab7.png)

**Comment by mikezimm** [comment]
## Logic to update html might be

1. Split string by <img
2. Starting at index 0 (but not the last one), add the "Marked verify div" to the ends of all strings.
3. Add <img back to the beginning of all splits starting at index 1 so html is not corrupt.
4. Then for each split starting at index 1:
5. Extract the source Url out of the image tag,
6. Insert Image web part after the previous text part
7. Then insert the next html string as text web part
8. Then do the same for the next image tag.

Basically do this:

---- start text web part
This is a bunch of text
`<div>.... Verify/Warn text`

---- insert image based on url from next split string

---- start text web part
This is a bunch of text
`<div>.... Verify/Warn text`

---- insert image based on url from next split string

**Comment by mikezimm** [comment]
## It turns out that the original html saves on creating the page...
But when you edit the page, SharePoint cleans up any unrecognized html including old image tags.
When saving, all the old html image tags are gone.

## Due to excessively complex html structure, I settled on the following
- Added toggle to markImagesAndLinks 
> This adds highlighted text above all image tags with the Url of the image.
> If you do edit the page, the highlighted text remains on the page so you can still know what did go there.

Added toggle to addImageWebParts
> This will add 2 web parts to the top of the page for every image.
> Webpart # 1 is the same warning tag with the image url
> Webpart # 2 is an image web part with the image url which.

When using this toggle, the intended update process would be:

1. Find highlighted text in the full text web part
2. Cut the page text above that highlighted text
3. Put that in the separate Text web part above the image.

It is a pain BUT what you do gain is:
You do not need to add a new Text web part and Image web part for every image on the page.

![image](https://user-images.githubusercontent.com/49648086/170805387-e52a0d4a-5f72-4b4b-b943-db4c796d263f.png)
